app/frontend/src/components/designer/FormDesigner.vue

```diff
@@ -382,6 +399,17 @@ export default {
         this.getFormSchema();
       }
     },
+    beforeDestroy() {
+      window.onbeforeunload = null;
+    }
+  },
+  watch: {
+    designerStep(newValue) {
+      if (newValue === 2) {
```

Comment:
wait, it does kick in, but only when i close the tab.  not leave the screen (ie navigate around in the app itself).

Reply:
This is a vue-router navigation issue. Will be looked at in future PR.